### PR TITLE
Bug: SLT calc again

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Fixed another SLT bug for a different type of broadcasting shape.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Ran locally where error was thrown and added more unit tests.

# Checklist:

- [N/A] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: pysat/utils/coords.py [hunk 1]
```diff
@@ -121,18 +121,30 @@ def calc_solar_local_time(inst, lon_name=None, slt_name='slt'):
...
+            coords.insert(0, inst.index.name)
 
-        slt = np.full(shape=sshape, fill_value=fill_val)
+            slt = np.full(shape=sshape, fill_value=fill_val)
 
-        # Calculate for each UT hr
-        for i, t in enumerate(ut_hr):
-            slt[i] = t + inst[lon_name] / 15.0
+            # Calculate for each UT hr
+            for i, hr in enumerate(ut_hr):
```
Comment (rstoneback) [inline pysat/utils/coords.py:146]: I missed the single letter variable name last time around. Good catch.

File: pysat/utils/coords.py [hunk 2]
```diff
@@ -121,18 +121,30 @@ def calc_solar_local_time(inst, lon_name=None, slt_name='slt'):
...
+                slt[i] = hr + inst[lon_name] / 15.0
 
     # Ensure that solar local time falls between 0 and 24 hours
     slt = np.mod(slt, 24.0)
```
Comment (rstoneback) [inline pysat/utils/coords.py:150]: Would you be willing to make this modulus optional? While I have things running for SAMI and the new multi-day thing, if the 24 hour modulus was optional I could probably use your function instead.

File: pysat/tests/test_utils_coords.py [hunk 3]
```diff
@@ -119,10 +119,12 @@ def test_bad_lon_name_calc_solar_local_time(self):
...
                                          slt_name='slt')
 
-    def test_lon_broadcasting_calc_solar_local_time(self):
+    @pytest.mark.parametrize("name", ["testmodel", "testing2d",
+                                      "testing2d_xarray"])
+    def test_lon_broadcasting_calc_solar_local_time(self, name):
         """Test calc_solar_local_time with longitude coordinates."""
 
-        self.py_inst = pysat.Instrument(platform='pysat', name="testmodel")
+        self.py_inst = pysat.Instrument(platform='pysat', name=name)
```
Comment (rstoneback) [inline pysat/tests/test_utils_coords.py:127]: Ran this manually on my end for additional confirmation. Also made a plot for the xarray case. Min and max check out as well.

test_plot

**Review by rstoneback — APPROVED** [review]

**Comment by aburrell** [comment]
Love the in-depth review, thanks!

File: pysat/utils/coords.py (hunk 2 above)
Comment (rstoneback) [reply pysat/utils/coords.py:150]: It was cute that I thought I get go through my first online exam without a barrage of emails :) Back on it now.

**Pull request merged**